lib/xopts.ts

```diff
@@ -0,0 +1,74 @@
+import {Signer} from 'ethers';
+import {providers} from 'ethers';
+import {ReadOnlyContracts, ReadWriteContracts} from './contracts';
+import {Addresses, Deployments} from './addresses';
+import {ExchangeRate, Currency, MonetaryAmount} from './monetary';
+
+type SignerOrProvider = Signer | providers.Provider;
+
+interface Option<From extends Currency, To extends Currency> {
+  expiry: Date;
+  from: From;
+  to: To;
+  strikePrice: ExchangeRate<From, To>;
+  currentPrice: ExchangeRate<From, To>;
```

Comment:
Same comment as above.

Additionally, this price actually updates depending on the amount you would want to exchange.